AutomatedTesting/Gem/PythonTests/terrain/EditorScripts/TerrainHeightGradientList_AddRemoveGradientWorks.py

```diff
@@ -0,0 +1,141 @@
+    )
+    terrain_data_changed_call_count_correct = (
+        "OnTerrainDataChanged called expected number of times.",
+        "OnTerrainDataChanged call count incorrect."
+    )
+
+def TerrainHeightGradientList_AddRemoveGradientWorks():
+    """
+    Summary:
+    Test aspects of the TerrainHeightGradientList through the BehaviorContext and the Property Tree.
+    :return: None
+    """
+
+    import os
+    import math as sys_math
+
+    import azlmbr.legacy.general as general
+    import azlmbr.bus as bus
+    import azlmbr.math as math
+    import azlmbr.terrain as terrain
+    import azlmbr.editor as editor
+    import azlmbr.vegetation as vegetation
+    import azlmbr.entity as EntityId
+
+    import editor_python_test_tools.hydra_editor_utils as hydra
+    from editor_python_test_tools.utils import Report
+    from editor_python_test_tools.utils import TestHelper as helper
+    import editor_python_test_tools.pyside_utils as pyside_utils
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+
+    terrain_changed_call_count = 0
+    expected_terrain_changed_calls = 0
+
+    aabb_component_name = "Axis Aligned Box Shape"
+    gradientlist_component_name = "Terrain Height Gradient List"
+    layerspawner_component_name = "Terrain Layer Spawner"
+
+    def create_entity_at(entity_name, components_to_add, x, y, z):
+        entity = EditorEntity.create_editor_entity_at([x, y, z], entity_name)
+        for component in components_to_add:
+            entity.add_component(component)
+
+        return entity
+        
+    def on_terrain_changed(args):
+        nonlocal terrain_changed_call_count
+
+        terrain_changed_call_count += 1
+
+    def set_component_path_val(entity, component, path, value):
+        components = entity.get_components_of_type([component])
+        components[0].set_component_property_value(path, value)
+
+    def set_gradients_check_height(main_entity, gradient_list, expected_height, test_results):
+        nonlocal expected_terrain_changed_calls
+
+        test_tolerance = 0.01
+        gradient_list_path = "Configuration|Gradient Entities"
+
+        set_component_path_val(main_entity, gradientlist_component_name, gradient_list_path, gradient_list)
+
+        expected_terrain_changed_calls += 1
+
+        # Wait until the terrain data has been updated.
+        helper.wait_for_condition(lambda: terrain_changed_call_count == expected_terrain_changed_calls, 2.0)
+
+        # Get the height at the origin.
+        height = terrain.TerrainDataRequestBus(bus.Broadcast, "GetHeightFromFloats", 0.0, 0.0, 0)
+
+        Report.result(test_results, sys_math.isclose(height, expected_height, abs_tol=test_tolerance))
+
+    helper.init_idle()
+
+    # Open a level.
+    helper.open_level("Terrain", "Base_WithTerrainWorldComponent")
+    helper.wait_for_condition(lambda: general.get_current_level_name() == "Base_WithTerrainWorldComponent", 2.0)
```

Comment:
It looks like there's still a new level being submitted with this automated test.  Is there any reason this is necessary, vs loading an existing empty test level and adding whatever is necessary to it programatically?